### PR TITLE
Dyn 5267 static splash screen

### PR DESCRIPTION
**Pull request by filipeotero**
This PR contains the static Splash Screen with mocked functions:

* Sign in
Opens a new window of the Autodesk login page

part1

* Launch Dynamo
Calls a method inside Dynamo to Launch it with a boolean that registers if the user wants to show this screen again

part2

* Import settings 
Reads a file
Send the string to Dynamo
Sets the status after the verification from Dynamo
The tooltip that shows the details of the error

part3

### Some important methods to be called:

//sets the data to be displayed in the loading bar
`window.setBarProperties("2.14","Launching", "100%", "Loading time: 100000ms")`

//Set the labels of the splash screen (for localization)
```
window.setLabels({
    welcomeToDynamoTitle: "Bem vindo ao Dynamo!",
    signInTitle: "Entrar",
    launchTitle: "Executar Dynamo 2.14",
    showScreenAgainLabel: "Não mostrar essa tela novamente"
})
```

//Called when the loading is finished
`window.setLoadingDone()`

//Set the status after importing the file
```
window.setImportStatus({
    status: 2,
    importSettingsTitle: 'Failed to Import Settings',
    errorDescription: 'Something went wrong when importing your custom setting file. Please try again or proceed with default settings.'
})
```
FYI: @QilongTang @reddyashish @zeusongit @RobertGlobant20 @jesusalvino

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -45,20 +51,40 @@ class App extends React.Component {
...
-  setWelcomeDynamoTitle(title) {
-    this.setState({ welcomeToDynamoTitle: title })
+  setLabels(labels) {
+    this.setState({
+      welcomeToDynamoTitle: labels.welcomeToDynamoTitle,
+      signInTitle: labels.signInTitle,
+      launchTitle: labels.launchTitle,
+      showScreenAgainLabel: labels.showScreenAgainLabel
+    })
+  }
```
Comment (QilongTang) [inline src/App.js:81]: Nice, can we add comments for this function to reflect the purpose. Please also add that in the future, devs should expand lables object to contain more resources

File: src/Dynamic.js [hunk 2]
```diff
@@ -8,7 +8,8 @@ class Dynamic extends React.Component {
         this.state = {
             barSize: "0%",
             dynamoVersion: "",
-            loadDescription: ""
+            loadDescription: "",
+            loadingTime: "Loading time: "
```
Comment (QilongTang) [inline src/Dynamic.js:12]: Do we need to localize this the same as we did to other resources?

File: src/Static.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+class Static extends React.Component {
+
+    constructor() {
+        super();
+
+        this.state = {
+            importStatus: importStatusEnum.none,
+            importSettingsTitle: "Import settings",
+            errorDescription: "Something went wrong when importing your custom setting file. Please try again or proceed with default settings."
```
Comment (QilongTang) [inline src/Static.js:28]: Similar question, should we also import these from somewhere?

**Review by QilongTang — COMMENTED** [review]
Other than a few questions on localization, looks good to me. Looking forward to have it integrated inside Dynamo

**Review by reddyashish — COMMENTED** [review]
LGTM

File: src/Static.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    //Opens a page to signin
+    signIn() {
+        window.open("https://accounts.autodesk.com/", "_blank");
+    }
+
+    //This method calls another method from Dynamo to actually launch it
+    launchDynamo() {
+        if (chrome.webview !== undefined) {
+            //The 'checked' is a boolean that represents if the user don't want to show the Static screen again
+            chrome.webview.hostObjects.scriptObject.LaunchDynamo(checked);
```
[the remark this answers is not in the document]
Comment (filipeotero) [reply src/Static.js:80]: Yes, that's exactly what I was planning. Adding a field into the settings that keeps the value if the user wants to show the screen again

File: src/Dynamic.js (hunk 2 above)
Comment (filipeotero) [reply src/Dynamic.js:12]: No need, it's part of setting the properties of the bar. IE:
`window.setBarProperties('2.14','Launching..','100%', 'Loading time: 1222222100ms')`

File: src/Static.js (hunk 3 above)
Comment (filipeotero) [reply src/Static.js:28]: It's also part of setting the state of the importation. 
```
window.setImportStatus({
    status: 2,
    importSettingsTitle: 'Failed to Import Settings',
    errorDescription: 'Something went wrong when importing your custom setting file. Please try again or proceed with default settings.'
})
```

**Review by QilongTang — APPROVED** [review]

File: src/Dynamic.js (hunk 2 above)
Comment (QilongTang) [reply src/Dynamic.js:12]: Gotcha

**Comment by QilongTang** [comment]
Although there will be more changes to login function which is worked on by @zeusongit , this is good to go. Nice work @filipeotero

**Pull request merged**